### ISSUE TITLE
PHP Warning: Attempt to read property post_parent in post-template.php on line 724

### ISSUE THREAD

**Issue by MarkRH** [stcr/subscribe-to-comments-reloaded#763]
I started noticing the following PHP Warning being generated in the server error log:

`[16-Jan-2023 03:19:28 UTC] PHP Warning: Attempt to read property "post_parent" on null in /(path)/wp-includes/post-template.php on line 724`

It turns out that it is being generated when I click on the **manage the subscriptions** link below a post or the **subscribe** link when I’m not logged in.

I am running **WordPress 6.1.1** with version **220725** of the plugin. I am also running using **PHP 8.1.14**. Looking at that line of code I could probably do a check beforehand to check for the existence of `$post->post_parent` in `if ( $post->post_parent )` before the if statement to skip it if it doesn’t.

Thought you’d like to know this is happening. Granted this is something in the main WordPress code but this plugin is somehow causing it to trigger.

I also happened to make a post here https://wordpress.org/support/topic/warning-attempt-to-read-property-post_parent-in-post-template-php-on-line-724/ and then realized you had GitHub page here.  

Thanks,
Mark H.

**Comment by BobaWebDev** [comment]
Hi @MarkRH 

Thanks for letting us know.

I'm not seeing on a fresh install (only WP and plugin).

Might be related to some other plugin trying to get the info about the current page (the STCR management page in this case) but since it's a fake page it's having trouble and showing the warning.

Can you send over a list of plugins you are using?

**Comment by MarkRH** [comment]
I have the following plugins installed:

- Advanced Editor Tools (previously TinyMCE Advanced)
- Akismet Anti-Spam
- Anti-spam Reloaded
- Asset CleanUp: Page Speed Booster
- CodeColorer ( I actually fixed its problem with PHP 8.1 )
- Disable Emojis (GDPR friendly)
- Disable Google Fonts
- Featured Image From URL
- Google Analytics for WordPress by MonsterInsights
- Lazy Load – Optimize Images
- PHP Everywhere
- Sassy Social Share
- Subscribe To Comments Reloaded
- WP-Polls

When I have time, I'll try disabling them all and add them one by one and see what happens but thought I'd give you this list for now.

Thanks,
Mark H.

**Comment by MarkRH** [comment]
I Deactivated all my plugins and it was still doing it.  Hmmm..

**Comment by BobaWebDev** [comment]
> I Deactivated all my plugins and it was still doing it. Hmmm..

Thanks for the info. 

Can you try creating a real page instead of a virtual page and check if the warning goes away?

- Go to WP Admin > StCR > Management Page
- Set “Virtual management page” to “No” and save changes
- Go to WP admin > Pages > Add New and create a page called `Comment subscriptions`, the permalink should be `comment-subscriptions`
- For the content of the page add `[stcr_management_page]`

**Comment by MarkRH** [comment]
I did that.  At first it didn't work but then I remembered I probably needed to put that code using a Shortcode black and not a Paragraph one..oops.  Now the page comes up in both scenarios (logged in and not logged in) without any warning being logged.

For what it's worth I am using an updated Twenty Twelve child theme.  I guess this can be marked as resolved since this work-around seems to be work now.
